ksqldb-engine/src/main/java/io/confluent/ksql/query/QueryRegistryImpl.java

```diff
@@ -231,9 +239,8 @@ public void restartStreamsRuntime(final KsqlConfig config) {
       final Map<String, Object> newStreamsProperties =
           new HashMap<>(config.getKsqlStreamConfigProps(applicationId));
       newStreamsProperties.put(StreamsConfig.APPLICATION_ID_CONFIG, applicationId);
-      final Map<String, Object> properties = QueryBuilder
-          .updateListProperties(newStreamsProperties);
-      stream.overrideStreamsProperties(properties);
+      QueryBuilder.updateListProperties(newStreamsProperties);
+      stream.overrideStreamsProperties(newStreamsProperties);
```

Comment:
note that previously, `properties` was the exact same object as `newStreamsProperties`.